packages/web-components/@sfx/product/src/variant.ts

```diff
@@ -0,0 +1,46 @@
+import { customElement, property, html } from 'lit-element';
+import { ProductVariantModel } from './product';
+
+import { Base } from '@sfx/base';
+
+@customElement('sfx-product-variant')
+export default class Variant extends Base {
+  @property({ type: String }) type: string = 'text';
+  @property({ type: Object }) variant: ProductVariantModel = { text: '', product: {} };
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    const { type, variant } = this;
+
+    // Configure variant component
+    this.setAttribute('role', 'listitem');
+    this.classList.add('product-variant');
+
+    switch (type) {
+      case 'color':
+        this.setAttribute('style', `background-color:${variant.color}`);
+        this.setAttribute('title', variant.text);
+        break;
+      case 'image':
+        this.setAttribute('style', `background-image:${variant.image};background-color:${variant.color}`);
+        this.setAttribute('title', variant.text);
+      case 'text':
+      default:
+        this.innerText = variant.text;
+    };
+
+    this.addEventListener('click', () => {
+      this.dispatchEvent(new CustomEvent('sfx::change_product_variant', {
```

Comment:
This is a reactive event, so it should be descriptive rather than imperative ("product\_variant\_change", for example). But take note of the comment above. As well, let's collect event names into a single file with exports so that it's easier to collect them into the new repo later on.